includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -867,29 +886,39 @@ private function process_css_list( CSSList $css_list, $options ) {
 	}
 
 	/**
-	 * Convert URLs in to non-relative real-paths.
+	 * Convert URLs in to non-relative normalized paths and prevent there being any spaces.
 	 *
 	 * @param URL[]  $urls           URLs.
-	 * @param string $stylesheet_url Stylesheet URL.
+	 * @param string $stylesheet_url Stylesheet URL. Base URL is obtained from this. Optional.
 	 */
-	private function real_path_urls( $urls, $stylesheet_url ) {
-		$base_url = preg_replace( ':[^/]+(\?.*)?(#.*)?$:', '', $stylesheet_url );
-		if ( empty( $base_url ) ) {
-			return;
-		}
+	private function normalize_urls( $urls, $stylesheet_url = '' ) {
```

Comment:
I think most of the changes to `normalize_urls` should be reverted back to `real_path_urls` because as we found out, the PHP CSS parser fails to parse URLs with spaces in them anyway, so this will never be fed any such URLs. The added comments should be retained, of course.

Reply:
Reverted all, except some new code comments.